### PR TITLE
Revealed intent with intermediate --test-framework option. Closes #96.

### LINKED ISSUE #96: Rename test option to test-framework
To avoid a confusion with the activity `test`ing we should prolly rename the long option to `testing-framework`, also aligns it with the used configuration key.

### PR DESCRIPTION
**Pull request by raphaelstolt**
Introduced intermediate `test-framework` option. Option `test` should be removed in a future release and the short option `t` should than be moved to the `test-framework`option.

### CODE REVIEW

File: src/Commands/ConstructCommand.php [hunk 1]
```diff
@@ -93,7 +93,8 @@ protected function configure()
         $this->setName('generate');
         $this->setDescription('Generates a basic PHP project');
         $this->addArgument('name', InputArgument::REQUIRED, $nameDescription);
-        $this->addOption('test', 't', InputOption::VALUE_OPTIONAL, $testDescription, 'phpunit');
+        $this->addOption('test', 't', InputOption::VALUE_OPTIONAL, $testFrameworkDescription, 'phpunit');
+        $this->addOption('test-framework', null, InputOption::VALUE_OPTIONAL, $testFrameworkDescription, 'phpunit');
         $this->addOption('license', 'l', InputOption::VALUE_OPTIONAL, $licenseDescription, 'MIT');
```
Comment (jonathantorres) [inline src/Commands/ConstructCommand.php:98]: The alias `test-framework` value is not being grabbed. When the command executes the value must be retrieved (if used) `$input->getOption('test-framework');` I think we should check both `test` and `test-framework` until the `test` option is removed on a major version.

**Comment by raphaelstolt** [comment]
Value of alias is now grabbed when not default (i.e. `phpunit`). We prolly should reduce the usage of magic variables (e.g. [phpunit](https://github.com/jonathantorres/construct/blob/master/src/Commands/ConstructCommand.php#L96)) by following the example of the [InteractiveCommand](https://github.com/jonathantorres/construct/blob/master/src/Commands/InteractiveCommand.php#L98) or adding constants to the [Defaults](https://github.com/jonathantorres/construct/blob/master/src/Defaults.php) class.

**Comment by jonathantorres** [comment]
The alias seems to work, but not the value. I tried running `construct generate jonathantorres/logger --test-framework=phpspec` and the project was generated, but using `phpunit`.

**Comment by raphaelstolt** [comment]
Works fine for me, tried it with phpspec, behat, and codeception. :confused:

**Comment by jonathantorres** [comment]
Working good now! Thanks!

**Pull request merged**